README.md

```diff
@@ -35,20 +35,94 @@ that can be used by other services, or natively interact with things such as GCP
 For events that include `data` Polyn also leverages the [JSON Schema](http://json-schema.org/)
 specification to create consistency.
 
-## Event and Data Serialization
+## Configuration
+
+### Domain
+
+The [Cloud Event Spec](https://github.com/cloudevents/spec/blob/v1.0.2/cloudevents/spec.md#type) specifies that every event "SHOULD be prefixed with a reverse-DNS name." This name should be consistent throughout your organization. You
+define that domain like this:
+
+```elixir
+config :polyn, :domain, "app.spiff"
+```
+
+### Event Source Root
+
+The [Cloud Event Spec](https://github.com/cloudevents/spec/blob/v1.0.2/cloudevents/spec.md#source-1) specifies that every event MUST have a `source` attribute and recommends it be an absolute [URI](https://en.wikipedia.org/wiki/Uniform_Resource_Identifier). Your application must configure the `source_root` to use for events produced at the application level. Each event producer can include its own `source` to append to the `source_root` if it makes sense.
+
+```elixir
+config :polyn, :source_root, "my_app"
+```
+
+### Connection
+
+You will need to provide the connection settings for your NATS connection. This will differ in-between environments.
+
+```elixir
+config :polyn, :nats, %{
+  name: :gnat, # (required) the registered named you want to give the Gnat connection
+  backoff_period: 4_000, # number of milliseconds to wait between consecutive reconnect attempts (default: 2_000)
+  connection_settings: [
+    %{host: "10.0.0.100", port: 4222},
+    %{host: "10.0.0.101", port: 4222},
+  ]
+}
+```
+
+## Event Schema Location
+
+Every Polyn Event must have a `dataschema` associated with it. Those JSON Schema files need to go in `priv/polyn/schemas`. Every event will be its own directory in the `schemas` folder without the `domain` prefix. Every event folder will have all the schema versions as json files inside it (also without the `domain` prefix). For example:
+
+```
+priv/
+├─ polyn/
+│  ├─ schemas/
+│  │  ├─ user.updated.v2/
```

Comment:
I'm going to modify this in the next PR once I move to an 1:1 event-schema model